drop-transfer/src/manager.rs

```diff
@@ -588,7 +594,11 @@ impl TransferManager {
             )
             .await;
         state.xfer_sync.local = sync::TransferState::Canceled;
-        let _ = state.conn.take();
+
+        if let Some(conn) = state.conn.take() {
+            let _ = conn.send(ServerReq::Close);
+        }
```

Comment:
What happens if `take()` returns `None` and how can it happen?

Reply:
It can happen when the peer is offline. Still, it's registered in the DB and is sent upon transfer resume